src/stargate.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{AcceptingModule, FailingModule, Module};
+use cosmwasm_std::{Binary, CosmosMsg, Empty, QueryRequest};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct StargateMsg {
+    pub type_url: String,
+    pub value: Binary,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct StargateQuery {
+    pub path: String,
+    pub data: Binary,
+}
+
+pub trait Stargate: Module<ExecT = StargateMsg, QueryT = StargateQuery, SudoT = Empty> {}
+
+pub type StargateAcceptingModule = AcceptingModule<StargateMsg, StargateQuery, Empty>;
+
+impl Stargate for StargateAcceptingModule {}
+
+pub type StargateFailingModule = FailingModule<StargateMsg, StargateQuery, Empty>;
+
+impl Stargate for StargateFailingModule {}
+
+impl<T> From<StargateMsg> for CosmosMsg<T> {
+    fn from(msg: StargateMsg) -> Self {
+        CosmosMsg::Stargate {
+            type_url: msg.type_url,
+            value: msg.value,
+        }
+    }
+}
```

Comment:
Do we need this conversion? Multitest handles a CosmosMsg by the contract, so StargateMsg -> CosmosMsg is something the contract does